### ISSUE TITLE
Support for live substitution

### ISSUE THREAD

**Issue by Akianonymus** [VonHeikemen/searchbox.nvim#22]
https://asciinema.org/a/92207

**Comment by VonHeikemen** [comment]
Tried to do that once, neovim would not let me modify the text of the buffer while the user is typing on the input.

**Comment by Akianonymus** [comment]
How about using a virtual text ? 🤔

**Comment by VonHeikemen** [comment]
like on top of the original text? Or what do you have in mind?

**Comment by Akianonymus** [comment]
> like on top of the original text? Or what do you have in mind?

Yeah, virtual text can write over original text, with highlights and such too.

Basically my target is to replicate live substitution like `: substitute` command.

**Comment by VonHeikemen** [comment]
can virtual text move the rest of the text in a "natural way"? If I change `awesome` with `awe` will the rest of the line adapt to that?

**Comment by Akianonymus** [comment]
Well, didn't think about it. I have a way, but it's kind of hacky.

Fetch the whole line, replace the text and then set that as virtual text as a whole line.

Probably highlights will be broken.

**Comment by Akianonymus** [comment]
Fair enough.
